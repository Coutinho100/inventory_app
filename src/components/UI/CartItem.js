import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
} from 'react-native';
import Card from '../Component/Card'
import { Entypo, FontAwesome5 } from '@expo/vector-icons';
const CartItem = props => {
    return (

        <View>
            <Card style={styles.cartItem}>
                <View style={styles.itemData}>
                    <FontAwesome5 name="capsules" size={24} color="black" />
                </View>
                <View>
                    <Text style={styles.mainText}>{props.title}</Text>
                    <Text style={styles.quantity}>Qty:{props.quantity} </Text>
                </View>
                <View style={styles.itemData}>
                    {/* <Text style={styles.mainText}>${props.amount.toFixed(2)}</Text> */}
                    {props.deletable && (
                        <TouchableOpacity
                            onPress={props.onRemove}
                            style={styles.deleteButton}
                        >
                            <Entypo name="cross" size={24} color="black" size={23}
                                color="red" />
                        </TouchableOpacity>
                    )}
                </View>
            </Card>
        </View>

    );
};

const styles = StyleSheet.create({
    cartItem: {
        padding: 10,
        backgroundColor: 'white',
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginHorizontal: 20,
        marginBottom: 20
    },
    itemData: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    quantity: {
        fontFamily: 'open-sans',
        color: '#888',
        fontSize: 16,
        paddingLeft: 10
    },
    mainText: {
        fontFamily: 'open-sans-bold',
        fontSize: 16
    },
    deleteButton: {
        marginLeft: 20
    }
});

export default CartItem;
