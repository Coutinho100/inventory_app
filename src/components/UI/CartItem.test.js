import React from 'react';
import CartItem from './CartItem';
import {
    configure,
    shallow
} from 'enzyme';
import Card from '../Component/Card'
import {
    Text,
} from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import test_function, {findByTestAttr} from '../../_test_/componenets/test_function'
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()



const dummy_data = {
    title:"hello",
    quantity:4,
    onRemove:function_click,
    deletable:false
}

let extra=
describe('DroneListItem />', () => {
  it("Events and props test in CartItem if deletable is false", () =>
        {
            wrapper = shallow(<CartItem  {...dummy_data} />)

            let maintext= findByTestAttr(wrapper,`mainText`)
            let quantity= findByTestAttr(wrapper,`quantity`)
            let touchComp= findByTestAttr(wrapper,`touchComp`)

            expect(maintext.props().children).toEqual("hello")
            expect(quantity.props().children).toEqual(['Qty:',4])
            expect(touchComp).toHaveLength(0);
            
            });
            it("Events and props test in CartItem if deletable is true", () =>
            {
                let new_prop=  {...dummy_data, deletable:true} ;
                wrapper = shallow(<CartItem {...new_prop}/>)
    
                let maintext= findByTestAttr(wrapper,`mainText`)
                let quantity= findByTestAttr(wrapper,`quantity`)
                let touchComp= findByTestAttr(wrapper,`touchComp`)
    
                expect(maintext.props().children).toEqual("hello")
                expect(quantity.props().children).toEqual(['Qty:',4])
                expect(touchComp).toHaveLength(1);
     
            touchComp.props().onPress()
            wrapper.update()
            expect(function_click).toHaveBeenCalledTimes(1)
        })
        });

test_function(["<CartItem/>",1,2],['renders correctly','renders the length of view'],[<CartItem />,Card,Text],extra)

