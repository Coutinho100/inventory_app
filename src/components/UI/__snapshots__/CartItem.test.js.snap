// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem/> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "padding": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
    </View>
    <View>
      <Text
        data-test="mainText"
        style={
          Object {
            "fontFamily": "open-sans-bold",
            "fontSize": 16,
          }
        }
      />
      <Text
        data-test="quantity"
        style={
          Object {
            "color": "#888",
            "fontFamily": "open-sans",
            "fontSize": 16,
            "paddingLeft": 10,
          }
        }
      >
        Qty:
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    />
  </View>
</View>
`;
